src/hardware/disk_noise.cpp

```diff
@@ -0,0 +1,467 @@
+			++spin.spin_it;
+		}
+	}
+
+	// Mix in seek sample, if it's playing
+	if (!seek.current_sample.empty() &&
+	    seek.current_it != seek.current_sample.end()) {
+		sample += (*seek.current_it) * DiskNoiseGain;
+		++seek.current_it;
+	}
+
+	// If seek sample has finished playing, clear entry
+	if (!seek.current_sample.empty() &&
+	    seek.current_it == seek.current_sample.end()) {
+		seek.current_sample.clear();
+		seek.current_it = seek.current_sample.end();
+	}
+
+	return AudioFrame{sample, sample};
+}
+
+void DiskNoiseDevice::LoadSample(const std::string& path, std::vector<float>& buffer)
+{
+	if (path.empty()) {
+		return;
+	}
+
+	constexpr auto SampleExtension = ".flac";
+
+	const auto candidate_paths = {std_fs::path(path),
+	                              std_fs::path(path + SampleExtension),
+	                              get_resource_path(DiskNoiseDir, path),
+	                              get_resource_path(DiskNoiseDir,
+	                                                path + SampleExtension)};
+
+	for (const auto& candidate : candidate_paths) {
+		if (!std_fs::exists(candidate)) {
+			continue;
+		}
+
+		// Load entire file into memory
+		std::ifstream file(candidate, std::ios::binary | std::ios::ate);
+		if (!file) {
+			LOG_WARNING("DISKNOISE: Failed to open file '%s'",
+			            candidate.c_str());
+			continue;
+		}
+
+		std::streamsize size = file.tellg();
+		file.seekg(0, std::ios::beg);
+
+		std::vector<uint8_t> file_data(static_cast<size_t>(size));
+		if (!file.read(reinterpret_cast<char*>(file_data.data()), size)) {
+			LOG_WARNING("DISKNOISE: Failed to read file '%s'",
+			            candidate.c_str());
+			continue;
+		}
+
+		// Decode FLAC from memory
+		drflac* decoder = drflac_open_memory(file_data.data(),
+		                                     file_data.size(),
+		                                     nullptr);
+		if (!decoder) {
+			LOG_WARNING("DISKNOISE: Failed to parse FLAC file '%s'",
+			            candidate.c_str());
+			continue;
+		}
+
+		const auto channels         = decoder->channels;
+		const auto sample_rate      = decoder->sampleRate;
+		const auto total_frames     = decoder->totalPCMFrameCount;
+		const auto HertzToKilohertz = 1000;
+
+		if (channels != 1) {
+			LOG_WARNING("DISKNOISE: FLAC file '%s' is not mono.",
+			            candidate.c_str());
+			drflac_close(decoder);
+			continue;
+		}
+		if (sample_rate != DiskNoiseSampleRateInHz) {
+			LOG_WARNING("DISKNOISE: FLAC file '%s' should be %dkHz, but %dkHz was found",
+			            candidate.c_str(),
+			            DiskNoiseSampleRateInHz / HertzToKilohertz,
+			            sample_rate / HertzToKilohertz);
+			drflac_close(decoder);
+			continue;
+		}
+
+		std::vector<float> temp(static_cast<size_t>(total_frames) * channels);
+		drflac_uint64 frames_read = drflac_read_pcm_frames_f32(decoder,
+		                                                       total_frames,
+		                                                       temp.data());
```

Comment:
You haven't started addressing linter warnings yet, so just a heads up. Probably you'll need to do something like this to make the linter happy:

```cpp
auto buffer_ptr = reinterpret_cast<float*>(buffer.data());
```

Then pass `buffer_ptr` in.